app/lib/features/pins/providers/notifiers/edit_state_notifier.dart

```diff
@@ -57,20 +60,46 @@ class PinEditNotifier extends StateNotifier<PinEditState> {
           hasChanges = true;
         }
       }
-      final content = pin.content()!;
-      if (content.body() != state.markdown) {
-        updateBuilder.contentMarkdown(state.markdown);
-        hasChanges = true;
+      final content = pin.content();
+      if (content != null) {
+        if (content.body() != state.markdown) {
+          updateBuilder.contentMarkdown(state.markdown);
+          hasChanges = true;
+        }
+
+        if (state.html != null) {
+          if (content.formattedBody() != state.html) {
+            updateBuilder.contentHtml(state.markdown, state.html!);
+            hasChanges = true;
+          }
+        }
       }
 
-      if (state.html != null && content.formattedBody() != state.html) {
-        updateBuilder.contentHtml(state.markdown, state.html!);
+      final client = ref.read(alwaysClientProvider);
+      final selectedAttachments = ref.read(selectedPinAttachmentsProvider);
```

Comment:
this all reads kind fragile ... not sure this is always updating exactly as we want it to...

Reply:
I don't have any thoughts how to approach this differently. Although I'm affirmative this works without breaking anything. Refer to the updated video on desktop flow!